File: clients/PHP-client/src/IndexesOpsClient.php
<?php

namespace SoliDB;

class IndexesOpsClient
{
    private Client $client;

    public function __construct(Client $client)
    {
        $this->client = $client;
    }

    public function rebuild(string $collection, string $indexName): void
    {
        $this->client->sendCommand('rebuild_index', [
            'database' => $this->client->getDatabase(),
            'collection' => $collection,
            'index_name' => $indexName
        ]);
    }

    public function rebuildAll(string $collection): void
    {
        $this->client->sendCommand('rebuild_all_indexes', [
            'database' => $this->client->getDatabase(),
            'collection' => $collection
        ]);
    }

    public function hybridSearch(string $collection, array $query): array
    {
        $args = array_merge([
            'database' => $this->client->getDatabase(),
            'collection' => $collection
        ], $query);
        $res = $this->client->sendCommand('hybrid_search', $args);
        return $res['data'] ?? [];
    }

    public function analyze(string $collection, string $indexName): array
    {
        $res = $this->client->sendCommand('analyze_index', [
            'database' => $this->client->getDatabase(),
            'collection' => $collection,
            'index_name' => $indexName
        ]);
        return $res['data'] ?? [];
    }

    public function getUsageStats(string $collection): array
    {
        $res = $this->client->sendCommand('index_usage_stats', [
            'database' => $this->client->getDatabase(),
            'collection' => $collection
        ]);
        return $res['data'] ?? [];
    }
}

File: clients/Ruby-client/lib/solidb/indexes_ops_client.rb
module SoliDB
  class IndexesOpsClient
    def initialize(client)
      @client = client
    end

    def rebuild(collection, index_name)
      @client.send_command("rebuild_index", database: @client.database, collection: collection, index_name: index_name)
      nil
    end

    def rebuild_all(collection)
      @client.send_command("rebuild_all_indexes", database: @client.database, collection: collection)
      nil
    end

    def hybrid_search(collection, query)
      args = { database: @client.database, collection: collection }.merge(query)
      @client.send_command("hybrid_search", args) || []
    end

    def analyze(collection, index_name)
      @client.send_command("analyze_index", database: @client.database, collection: collection, index_name: index_name)
    end

    def get_usage_stats(collection)
      @client.send_command("index_usage_stats", database: @client.database, collection: collection)
    end
  end
end
